djangocms_googlemap/templates/cms/plugins/googlemap.html

```diff
@@ -1,75 +1,87 @@
 {% load i18n sekizai_tags %}
 
 {% addtoblock "js" %}
-  <script type="text/javascript">
-function initializeMap{{ object.pk}}() {
-      // save settings form plugin
-      var container = 'google-map-{{ object.pk }}';
-      var latlng = new google.maps.LatLng(46.94708, 7.445975); // default to switzerland
-
-      var title = "{{ object.get_title }}";
-      var address = "{{ object.address }}, {{ object.zipcode }} {{ object.city }}";
+<script async defer src="https://maps-api-ssl.google.com/maps/api/js?v=3&amp;callback=initializeMaps"></script>
+{% endaddtoblock %}
 
-      var options = {
-        zoom: {{ object.zoom }},
-        mapTypeId: google.maps.MapTypeId.ROADMAP, /* ROADMAP, SATELLITE, HYBRID or TERRAIN */
+{% addtoblock "js" %}
+<script type="text/javascript">
+    if (typeof _djangocms_googlemap_mapsToInitialize === 'undefined') {
+        _djangocms_googlemap_mapsToInitialize = {};
+    }
+    _djangocms_googlemap_mapsToInitialize[{{ object.pk }}] = {
+        zoom: {{ object.zoom }},
         scrollwheel: {{ object.scrollwheel|lower }},
         disableDoubleClickZoom: {% if object.double_click_zoom %}false{% else %}true{% endif %},
-        draggable: {{ object.draggable|lower}},
-        keyboardShortcuts: {{ object.keyboard_shortcuts|lower}},
-        panControl: {{ object.pan_control|lower}},
-        zoomControl: {{ object.zoom_control|lower}},
-        streetViewControl: {{ object.street_view_control|lower}}
-      };
-
-      {% if object.get_lat_lng %}
-        // load latlng from given lat lng settings
-        latlng = new google.maps.LatLng({{ object.get_lat_lng|join:", " }});
-        initGoogleMaps(latlng);
-      {% else %}
-        // load latlng from given address
-        var geocoder = new google.maps.Geocoder();
-          geocoder.geocode({'address': address}, function (results, status) {
-            // check if address can be found if not show default (var latlng)
-            if(status == google.maps.GeocoderStatus.OK) latlng = results[0].geometry.location;
-            initGoogleMaps(latlng);
-          });
-      {% endif %}
+        draggable: {{ object.draggable|lower }},
+        keyboardShortcuts: {{ object.keyboard_shortcuts|lower }},
+        panControl: {{ object.pan_control|lower }},
+        zoomControl: {{ object.zoom_control|lower }},
+        streetViewControl: {{ object.street_view_control|lower }},
+        title: "{{ object.get_title }}",
+        address: "{{ object.address }}, {{ object.zipcode }} {{ object.city }}",
+        lat: {% if object.lat != None %}{{ object.lat }}{% else %}null{% endif %},
+        lng: {% if object.lng != None %}{{ object.lng }}{% else %}null{% endif %},
+        info_window: {{ object.info_window|lower }},
```

Comment:
not 100% sure here if this can be `None`, but if it is then it's better to follow similar approach as with `disableDoubleClickZoom` with an `if` statement